File: AppForCustomers/src/core/classes/menu/order.ts
import { OrderStatus } from "./orderStatus";
import { OrderDetail } from "./orderDetail";
import { Table } from "./table";

export class Order 
{
    constructor(public id: number, 
        public table: Table, 
        public orderStatus: OrderStatus,
        public orderDetails: Array<OrderDetail>,
        public total: number)
        {}
}

File: src/Core/Domains/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace Domains.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public Table Table { get; set; }
        public OrderStatus OrderStatus { get; set; }
        public List<OrderDetails> OrderDetails { get; set; }  
        public decimal Total { get; set; }
    }
}